packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -222,7 +222,7 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           // for relative js/css imports, inherit importer's version query
           // do not do this for unknown type imports, otherwise the appended
           // query can break 3rd party plugin's extension checks.
-          if (isRelative && !/[\?&]import\b/.test(url)) {
+          if (isRelative && !/[\?&]import=vitedev\b/.test(url)) {
```

Comment:
Good catch, interesting that we didn't get more reports about this before.

The fix looks good to me, but I think that we should use `import=true` for the value.

Reply:
Sounds like a plan to me :)

I will switch this over to just use the `=`

Side note: it was a little rough identifying all the cases where `import` existed and was a query flag. IMO, it would be really nice if that flag was named without a common keyword. I added the `vitedev` value both to make CMD+F easier and to help future devs identify similar bugs in the future.

Not planning on making any further changes atm, but I wanted to raise this as I wrap up. Ty again for the feedback and help!!

Reply:
I agree, there are some projects in the ecosystem that are using this feature even if it is internal so we need to be careful to change it. The issue is also that if we prefix everything with something like `vite_import`, `vite_worker`, etc, then the URL becomes quite noisy.